### ISSUE TITLE
Blitzkrieg Anthology (313480)

### ISSUE THREAD

**Issue by IanTrudel** [ValveSoftware/Proton#3331]
# Compatibility Report
- Name of the game with compatibility issues: Blitzkrieg Anthology
- Steam AppID of the game: 313480

## System Information
- GPU: NVIDIA GeForce GTX 660M
- Driver/LLVM version: NVIDIA 435.27.07
- Kernel version: 5.3.12
- Link to full system information report as [Gist](https://gist.github.com/):
- Proton version: 4.11-10 (not working), 4.2-9 (working)

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

[steam-313480.log](https://github.com/ValveSoftware/Proton/files/3965918/steam-313480.log)

## Symptoms 

The game will not work with the newest version of Proton, namely 4.11-10, but will work with version 4.2-9. As a side note, it seems to work better with `PROTON_USE_D9VK=1 %command%` on the latter version.

## Reproduction

1. Launch the game.

**Comment by alasky17** [comment]
@BackOrder We are not seeing this failure, with or without d9vk.  I assume that the failure happens right on launch, not once you get into a campaign?

Could you try switching back to 4.11-9 to see this can be narrowed down to the upgrade from -9 to -10?  You can switch by going into the properties for Proton 4.11-10, going to the beta tab, and selecting proton-4.11-9.

Also, could you try without d9vk as well as with d9vk?

**Comment by IanTrudel** [comment]
Hi @alasky17 !

I did try and retry Proton version 4.2-9, 4.11-9 and 4.11-10 with and without d9vk. The results are not overly reliable when it comes to 4.11* but 4.2-9 works best. To test, I recommend you to go to **New Game > Tutorials**, continue. You should see a tank appear at the middle-bottom of the screen (on the road). 4.11* just crash instead of displaying the tank.

It seems to be a resolution problem because previous version of Proton would list more resolutions than defined on my desktop (see xrandr output below). The newest version of Proton only list one resolution, same as my desktop. Same goes with some Unity games.

Output from xrandr:
```
LVDS-0 connected primary 1920x1080+0+0 (normal left inverted right x axis y axis) 382mm x 215mm
   1920x1080     60.01 +  40.01* 
```

Darkest Hour ([73170](https://github.com/ValveSoftware/Proton/issues/3338)) also demonstrate the same resolution problem where 4.2-9 will list multiple resolutions but 4.11* will only list 1920x1080. Might be related.

**Comment by alasky17** [comment]
@BackOrder 

Thank you for the investigation!  Your log had the following lines, which seem to be the culprit: 
`err:winediag:xrandr12_init_modes Broken NVIDIA RandR detected, falling back to RandR 1.0. Please consider using the Nouveau driver instead.
1046716.288:0028:0029:err:xrandr:xrandr10_init_modes xrandr 1.2 support required`

We are going to remove the code that was triggered there (you can check the changelog to see when the change happens in Proton if I forget to update this for some reason).  Hopefully that should get you back up and running on the 4.11 branch.

It is interesting because on most setups, Blitzkrieg (and probably the others, but that is the one I checked) lists more resolutions than just ones from xrandr.  Hence why no one realized this was a problem until now :)

**Comment by IanTrudel** [comment]
Fantastic @alasky17 , you are so fast! :)

**Comment by alasky17** [comment]
@BackOrder The change isn't in Proton 4.11-11 ... We looked into it after -11 was already in process, but it is on the list :)

**Comment by aeikum** [comment]
@BackOrder Hopefully this will be working again in Proton 4.11-12. Please retest.

**Comment by IanTrudel** [comment]
@aeikum no cake! It doesn't work. Now there are a lot of the following:

`56438.967:0026:0027:err:x11settings:X11DRV_Settings_AddOneMode Maximum modes (16) exceeded`

[steam-313480.log](https://github.com/ValveSoftware/Proton/files/4066104/steam-313480.log)

**Comment by aeikum** [comment]
Huh. Can you add `+x11settings` to the WINEDEBUG line and upload another log?

**Comment by IanTrudel** [comment]
Technologies are fascinating! :) [steam-313480.log](https://github.com/ValveSoftware/Proton/files/4066216/steam-313480.log)

**Comment by aeikum** [comment]
@BackOrder Thanks, I think I have a fix for this.

Are you comfortable with a command line? Here's some instructions to try it out if so.

Download this archive: [issue3331.tar.gz](https://github.com/ValveSoftware/Proton/files/4066325/issue3331.tar.gz)

Head into your Proton 4.11 directory, something like this (I noticed you have a different Steam library, so your path may be different):

    cd ~/.steam/steam/steamapps/common/Proton\ 4.11

Then extract the archive there:

    tar -xvf ~/Downloads/issue3331.tar.gz

This should overwrite two winex11.drv.so files inside the `dist` folder.

Then try the game again and see if it is fixed.

**Comment by IanTrudel** [comment]
You are a master at your craft. The game is working and it is now possible to change the resolution in-game.

I also tried a Windows/Unity game for good measure and the ScreenResolution window did list a whole bunch of resolutions. So far so good.

There are exceptions raised in the following log but they are unrelated to the resolution issue and the game is playable.

[Uploading steam-313480.log…]()

**Comment by aeikum** [comment]
Great, thanks for testing! That fix will be in the next Proton release.

**Comment by aeikum** [comment]
Hey @BackOrder, looks like I forgot to update this. This fixed was shipped in 5.0-1 and in 4.11-13. Thanks very much for reporting and for your time testing out the fix. OK to close this?
